### ISSUE TITLE
Language support matrix

### ISSUE THREAD

**Issue by codebrainz** [geany/www.geany.org#4]
I guess this is where to put this idea for discussion :)

It might be nice to have a table on the website which has rows for all supported languages and has a column for each of supported features (ex. symbols, syntax highlighting, is "custom" filetype?, etc). I thought about stuffing this on the wiki, but it seems like useful website content that people evaluating Geany would expect to find.

Some random Google results for "language support matrix" for similar to what I'm talking about are [here](https://www.relativity.com/relativity/Portals/0/Documents/7.5%20Documentation%20Help%20Site/Content/Relativity%20Administrator/Supported%20languages%20matrix/Supported%20languages%20matrix.htm), [here](https://thrift.apache.org/docs/Languages), [here](https://w3c.github.io/typography/gap-analysis/language-matrix.html), and [here](https://langserver.org/).

What do you think?

What else could go in the columns in addition to the examples above?

@eht16 What would be the best format to make the table in for the new website?

**Comment by eht16** [comment]
Good idea.

We already have the plain list of supported filetypes/languages in https://github.com/geany/www.geany.org/blob/master/page_content/about/filetypes.md.
I guess we can replace the plain list with a matrix on that page.

I'd suggest Markdown tables as format, like this:
```
Language       | Highlighting  | Symbol parsing | Highlighting | Installation
-------------- | ------------- | -------------- | ------------ | -----------
C              | &#x2611;      | &#x2611;       | &#x2611;     | builtin
C++            | &#x2611;      | &#x2611;       | &#x2611;     | builtin
Dockerfile     | &#x2611;      | &#x2611;       | &#x2611;     | [Wiki page](https://wiki.geany.org/config/dockerfile)
Lilypong       | &#x2611;      | &#x2611;       | &#x2611;     | [Wiki page](https://wiki.geany.org/config/lilypond)
Python         | &#x2611;      | &#x2611;       | &#x2611;     | builtin
YAML           | &#x2611;      | &#x2610;       | &#x2611;     | builtin
```

The Unicode HTML entities are rendered as checked and unchecked boxes.

**Comment by codebrainz** [comment]
Cool, I will try and spend some time starting such a table.

Are you sure the Markdown processor used by the website supports that extension for tables?

**Comment by b4n** [comment]
Sounds good indeed :)

**Comment by eht16** [comment]
> Are you sure the Markdown processor used by the website supports that extension for tables?

Yes, we already use the extension on the Download->Releases page and it has been therefore enabled in settings.py:
https://github.com/geany/www.geany.org/blob/master/geany/settings.py#L374

I just added some basic CSS formatting of plain tables in cb57404f01b66eb68d2052b9a79613ced363f224. So it should pretty good by default.
